functions.sh

```diff
@@ -1,18 +1,31 @@
 #!/usr/bin/env bash
 
+
+# Determine Dockerhub name by splitting the "repo slug",
+# i.e. if the GitHub repo is Silex/emacs then
+# DOCKER_REPO=silex
+
+DOCKER_REPO=$(echo $TRAVIS_REPO_SLUG | cut -d "/" -f 1 | tr '[:upper:]' '[:lower:]')
+DOCKER_IMG=emacs
+
 build()
 {
-  for tag in $DOCKER_TAGS; do
-    docker build -t silex/emacs:$tag --build-arg="GIT_BRANCH=$GIT_BRANCH" .
-  done
+    for tag in $DOCKER_TAGS; do
+        major=$(echo $tag | cut -d "." -f 1)  # i.e. "24.5" => "25"
+        echo Build $DOCKER_REPO/$DOCKER_IMG:$tag from source $GIT_BRANCH
+        docker build -t $DOCKER_REPO/$DOCKER_IMG:$tag \
+               --build-arg="GIT_BRANCH=$GIT_BRANCH" \
+               -f Dockerfile.$major .
+    done
 }
 
 push()
 {
-  if [ "$TRAVIS_BRANCH" == "master" ]; then
-    docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
-    for tag in $DOCKER_TAGS; do
-      docker push silex/emacs:$tag
-    done
-  fi
+    if [ "$TRAVIS_BRANCH" == "master" ]; then
+        docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
+        for tag in $DOCKER_TAGS; do
+            echo Push $DOCKER_REPO/$DOCKER_IMG:$tag
+            docker push $DOCKER_REPO/$DOCKER_IMG:$tag
+        done
+    fi
```

Comment:
Looks like you use intendation at 4 spaces. This project used 2 spaces until now...